src/files/BinaryFile.cpp

```diff
@@ -336,14 +366,43 @@ void BinaryFile::seek(uint64_t position) {
 #endif
 /***** adapted from boost ******/
 
-inline uint16_t swap_u16(uint16_t value) {
+template<size_t Size>
+struct unsigned_type;
+
+template<>
+struct unsigned_type<2>{
+    typedef uint16_t type;
+};
+
+template<>
+struct unsigned_type<4>{
+    typedef uint32_t type;
+};
+
+template<>
+struct unsigned_type<8>{
+    typedef uint64_t type;
+};
+
+template<typename T>
+inline T swap_endianness(T value) {
+    typename unsigned_type<sizeof(T)>::type uval;
+    std::memcpy(&uval, &value, sizeof(T));
+    uval = swap_endianness(uval);
+    std::memcpy(&value, &uval, sizeof(T));
+    return value;
```

Comment:
Also, this might be worth adding a `static_assert<std::is_arithmetic<T>::value>` just in case someone tries to use this function with a `struct {uint8_t a; uint8_t b;}` at some point.

Reply:
Added the assert and the comment.